### PR TITLE
DOCS-845: Updates for MinIO Server RELEASE.2023-05-04T21-44-30Z

### LINKED ISSUE #845: [RELEASE] MinIO RELEASE.2023-05-04T21-44-30Z doc changes
MinIO [RELEASE.2023-05-04T21-44-30Z](https://github.com/minio/minio/releases/tag/RELEASE.2023-05-04T21-44-30Z) has changes that impact the docs:

- [x] `root` can now be disabled in config settings ([PR #17089](https://github.com/minio/minio/pull/17089))
- [x] Support for systemd notification protocol ([PR #17062](https://github.com/minio/minio/pull/17062))
- [x] Directory objects no longer create new versions ([PR #17109](https://github.com/minio/minio/pull/17109))
- [x] Batch replication supports arbitrary s3 source or target ([PR #17113](https://github.com/minio/minio/pull/17113))
- [ ] Added support for virtual routing and forwarding on Linux ([17108](https://github.com/minio/minio/pull/17108))
- [x] New environment variable to control console animation ([PR #17123](https://github.com/minio/minio/pull/17123))
        See also [Console PR #2799](https://github.com/minio/console/pull/2799)
- [x] Allow directory deletes to perform permanent deletes ([PR #17132](https://github.com/minio/minio/pull/17132))
- [x] EC:0 value not allowed except at server startup ([PR #17141](https://github.com/minio/minio/pull/17141))

### PR DESCRIPTION
**Pull request by ravindk89**
Closes #845 

Staged: http://192.241.195.202:9000/staging/DOCS-845/linux/index.html

### CODE REVIEW

File: source/includes/common-mc-admin-config.rst [hunk 1]
```diff
@@ -1007,3 +1007,20 @@ Specify the maximum limit for undelivered messages. Defaults to ``100000``.
...
 .. end-minio-notify-webhook-comment
+
+.. Root API Access
+
+.. start-minio-root-api-access
+
+.. versionadded:: MinIO Server RELEASE.2023-05-04T21-44-30Z
+
+Specify ``on`` to enable and ``off`` to disable the :ref:`root <minio-users-root>` user account.
+Disabling the root service account also disables all service accounts associated with root, excluding those used by site replication.
```
Comment (ravindk89) [inline source/includes/common-mc-admin-config.rst:1018]: @harshavardhana are there are any other internal services which would continue to support root and/or its service accounts beyond site replication?

File: source/includes/linux/common-installation.rst [hunk 2]
```diff
@@ -233,6 +233,11 @@ file manually on all MinIO hosts:
    ExecStartPre=/bin/bash -c "if [ -z \"${MINIO_VOLUMES}\" ]; then echo \"Variable MINIO_VOLUMES not set in /etc/default/minio\"; exit 1; fi"
    ExecStart=/usr/local/bin/minio server $MINIO_OPTS $MINIO_VOLUMES
 
+   # MinIO RELEASE.2023-05-04T21-44-30Z adds support for Type=notify (https://www.freedesktop.org/software/systemd/man/systemd.service.html#Type=)
+   # This may improve systemctl setups where other services use `After=minio.server`
+   # Uncommon the line to enable the functionality
```
Comment (feorlen) [inline source/includes/linux/common-installation.rst:238]: `uncomment` I presume. Is this example from a code repo or somewhere else that might further propagate a typo?

File: source/reference/minio-server/minio-server.rst [hunk 3]
```diff
@@ -445,6 +450,19 @@ MinIO Console:
    the external identity provider has a reachable URL to which to send the
    authentication response.
 
+.. envvar:: MINIO_SERVER_URL
+
+   *Optional*
+
+   Specify the URL hostname the MinIO Console should use for connecting to the
```
Comment (feorlen) [inline source/reference/minio-server/minio-server.rst:457]: Perhaps `to connect`?

Is this an arbitrary URL, or does it need to be just the domain name? `URL hostname` sounds like you can't include a path.

Also should join the broken sentences in this section. (I'm guessing this text came from elsewhere.)

File: source/reference/minio-server/minio-server.rst [hunk 4]
```diff
@@ -312,6 +312,15 @@ Root Credentials
...
       :envvar:`MINIO_ROOT_PASSWORD` value for all environments.
 
+.. envvar:: MINIO_API_ROOT_ACCESS
+
+   .. include:: /includes/common-mc-admin-config.rst
+      :start-after: start-minio-root-api-access
+      :end-before: end-minio-root-api-access
+
+   This variable corresponds to the :mc-conf:`api root_access <api.root_access>` configuration setting.
+   You can use this variable to temporarily override the configuration setting and re-enable root access to the deployment.
```
Comment (djwfyi) [inline source/reference/minio-server/minio-server.rst:322]: Might be worth mentioning that envvar overrides config setting.

 Both on this page and the `mc admin config` page, I would think.

File: source/reference/minio-server/minio-server.rst [hunk 5]
```diff
@@ -522,6 +540,16 @@ refers to the specific storage tier on which to store a given object.
    This environment variable only applies to deployments with
    :ref:`Erasure Coding <minio-erasure-coding>` enabled.
 
+   The minimum value at startup is ``0``.
+   0 parity setups have no erasure coding protections and rely entirely on the storage controller or resource for availability / resiliency. 
+
+   The maximum value is 1/2 the erasure set stripe size.
+   For example, a deployment with erasure set stripe size of 16 has a maximum standard parity of 8.
+
+   You can change the Standard parity after startup to a value between ``1`` and ``1/2(ERASURE_SET_SIZE)``.
```
Comment (djwfyi) [inline source/reference/minio-server/minio-server.rst:549]: Can use the `math` role here:

```suggestion
   You can change the Standard parity after startup to a value between ``1`` and :math:`\tfrac{1}{2}\ (ERASURE_SET_SIZE)`.
```

https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-math
https://ctan.mirrors.hoobly.com/info/short-math-guide/short-math-guide.pdf

File: source/reference/minio-server/minio-server.rst [hunk 6]
```diff
@@ -522,6 +540,16 @@ refers to the specific storage tier on which to store a given object.
...
+   MinIO only applies the changed parity to newly written objects.
+   MinIO does not apply the new parity value to existing objects.
```
Comment (djwfyi) [inline source/reference/minio-server/minio-server.rst:551]: Perhaps
```suggestion
Existing objects retain the parity value in place at the time of their creation.
```

**Review by djwfyi — CHANGES_REQUESTED** [review]
A few suggestions to consider.
I'll take another look after the next pass.

File: source/includes/linux/common-installation.rst (hunk 2 above)
Comment (ravindk89) [reply source/includes/linux/common-installation.rst]: No this is from me :grimacing:

File: source/reference/minio-server/minio-server.rst (hunk 4 above)
Comment (ravindk89) [reply source/reference/minio-server/minio-server.rst:322]: Done

File: source/reference/minio-server/minio-server.rst (hunk 3 above)
Comment (ravindk89) [reply source/reference/minio-server/minio-server.rst]: FQDN now that I think about it.

Comment (ravindk89) [reply source/reference/minio-server/minio-server.rst]: Also @harshavardhana @kannappanr i feel like we improved this behavior sufficiently in a recent release, but I can't find the PR.

File: source/reference/minio-server/minio-server.rst (hunk 5 above)
Comment (ravindk89) [reply source/reference/minio-server/minio-server.rst]: Thank you for putting the effort into this so I dont have to

**Review by ravindk89 — COMMENTED** [review]
Back to y'all for more

File: source/reference/minio-mc-admin/mc-admin-config.rst [hunk 7]
```diff
@@ -102,6 +102,9 @@ Root User Account
          :end-before: end-minio-root-api-access
 
       This configuration setting corresponds with the :envvar:`MINIO_API_ROOT_ACCESS` environment variable.
+      To reset after unintentional locket, set :envvar:`MINIO_API_ROOT_ACCESS` ``on`` to override this setting and temporarily re-enable the root account.
```
Comment (feorlen) [inline source/reference/minio-mc-admin/mc-admin-config.rst:105]: `locket` presumably was intended to be `lock`?

Comment (feorlen) [reply source/reference/minio-mc-admin/mc-admin-config.rst:105]:
```suggestion
      To reset after unintentional lock, set :envvar:`MINIO_API_ROOT_ACCESS` ``on`` to override this setting and temporarily re-enable the root account.
```

**Review by feorlen — APPROVED** [review]
suggested a typo fix, everything else is 👍🏻

**Review by djwfyi — APPROVED** [review]
One indent question, otherwise LGTM.

File: source/reference/minio-server/minio-server.rst [hunk 8]
```diff
@@ -546,9 +542,9 @@ refers to the specific storage tier on which to store a given object.
    The maximum value is 1/2 the erasure set stripe size.
    For example, a deployment with erasure set stripe size of 16 has a maximum standard parity of 8.
 
-   You can change the Standard parity after startup to a value between ``1`` and ``1/2(ERASURE_SET_SIZE)``.
+   You can change the Standard parity after startup to a value between ``1`` and :math:`\tfrac{1}{2}\ (ERASURE_SET_SIZE)`.
    MinIO only applies the changed parity to newly written objects.
-   MinIO does not apply the new parity value to existing objects.
+Existing objects retain the parity value in place at the time of their creation.
```
[the remark this answers is not in the document]
Comment (ravindk89) [reply source/reference/minio-server/minio-server.rst:547]: The entire suggestion is gone on my end. Odd.

**Pull request merged**